jerry-core/ecma/base/ecma-property-hashmap.c

```diff
@@ -492,6 +485,62 @@ ecma_property_hashmap_find (ecma_property_hashmap_t *hashmap_p, /**< hashmap */
   uint32_t start_entry_index = entry_index;
 #endif /* !JERRY_NDEBUG */
 
+  if (ECMA_IS_DIRECT_STRING (name_p))
+  {
+    ecma_property_t prop_name_type = (ecma_property_t) ECMA_GET_DIRECT_STRING_TYPE (name_p);
+    jmem_cpointer_t property_name_cp = (jmem_cpointer_t) ECMA_GET_DIRECT_STRING_VALUE (name_p);
+
+    JERRY_ASSERT (prop_name_type > 0);
+
+    while (true)
+    {
+      if (pair_list_p[entry_index] != ECMA_NULL_POINTER)
+      {
+        size_t offset = 0;
+        if (ECMA_PROPERTY_HASHMAP_GET_BIT (bits_p, entry_index))
+        {
+          offset = 1;
+        }
+
+        ecma_property_pair_t *property_pair_p = ECMA_GET_NON_NULL_POINTER (ecma_property_pair_t,
+                                                                           pair_list_p[entry_index]);
+
+        ecma_property_t *property_p = property_pair_p->header.types + offset;
+
+        JERRY_ASSERT (ECMA_PROPERTY_IS_NAMED_PROPERTY (*property_p));
+
+        if (property_pair_p->names_cp[offset] == property_name_cp
+            && ECMA_PROPERTY_GET_NAME_TYPE (*property_p) == prop_name_type)
+        {
+#ifndef JERRY_NDEBUG
```

Comment:
Why do we need the `#ifndef` here? (and at the other places)

Reply:
Because asserts implemented in a interesting way and references its variables without evaluating them in release mode.